myfitnesspal/client.py

```diff
@@ -587,7 +590,7 @@ def set_measurements(
 
         # setup a dict for the post
         data = {
-            "authenticity_token": self._authenticity_token,
+            "authenticity_token": authenticity_token,
```

Comment:
Sorta curious if you have any insight into why `self._authenticity_token` isn't set in the first place?  It seems like such a bug might be breaking _other_ things, too, and it might be wise to fix that, instead, no?

Reply:
from what i understand, the commit my PR is based off just doesn't parse the `authenticity_token` particular to this request.

the only request in the chain of invoking/initializing the library is when you log in with your credentials, which stores to `self._authenticity_token` a token that will work fine for getting various values, but not for setting values (which as i alude to in the OP appears to require function-/request-specific `authenticity_token`s).

as such, i would regard this token as a session-bound token that is entirely separate from the authenticity token required for each call to a value-storing option. this token changes everytime you open the page containing the form to add values, so this value truly is local to the function call, cannot be reused and is necessary to make a successful call.

thus, this implementation (in a very general sense) seems necessary.